nncf/experimental/common/pruning/propagation_data.py

```diff
@@ -11,24 +11,54 @@
  limitations under the License.
 """
 
+from dataclasses import dataclass
 import json
-from typing import Dict
+from typing import Dict, Optional, Set
 from typing import List
 
+@dataclass
+class ProducerInfo:
+    """
+    Defines a node that produce pruning. Operation that corresponds to the node has built-in weights.
```

Comment:
```suggestion
    Defines a node that produce pruning masks. Operation that corresponds to the node has built-in weights.
```

Reply:
Originally, I wanted to say in general about removing elements, without implementation details with masks. my version looks better for me.

Reply:
Discussed in person, corrected